fabtests/common/shared.c

```diff
@@ -2659,33 +2659,33 @@ int ft_sync()
 
 	if (opts.dst_addr) {
 		if (!(opts.options & FT_OPT_OOB_SYNC)) {
-			ret = ft_tx(ep, remote_fi_addr, 1, &tx_ctx);
+			ret = ft_tx(ep, remote_fi_addr, 0, &tx_ctx);
 			if (ret)
 				return ret;
 
 			ret = ft_rx(ep, 1);
 		} else {
-			ret = ft_sock_send(oob_sock, &buf, 1);
+			ret = ft_sock_send(oob_sock, &buf, 0);
```

Comment:
Can you update the commit message to say why this change is made?  Note the receiving 0-bytes makes it impossible to know if recv returned 0 because that's how many bytes were requested or the connection was torn down.  I also don't know what the behavior is if recv is called with 0-bytes and there's no data to be read because the send hasn't happened.  Did you confirm this sort of sync still works?

Reply:
Sure, I can update the commit message with something short.  More detail for you here:

I'm working on making a version of fi_rdm_atomic that validates the atomic operation results in the expected answer so that we can gain confidence in some of the recent HMEM atomics we've been working on in prov/efa.  I realized the current ft_sync implementation was incompatible because the tx and rx modify the (global) buffers invalidating the check.


I don't actually need to change the `ft_sock_send` lines, because those operate on that local `buf` instead of the rx_buf/tx_buf.  I'll put those back to 1 byte to avoid the recv confusion.  After testing the out-of-band sync I think you are right that recv isn't really doing anything.